main/main.c

```diff
@@ -102,6 +111,35 @@ parse_opt(int key, char *arg, struct argp_state *state)
 		args->log_base_dir = arg;
 		break;
 
+	case 'm': {
+		long log_file_mode;
+		char *end;
+
+		/* Assuming the file mode is in octal. */
+		log_file_mode = strtol(arg, &end, 8);
+		if (arg == end || !*arg || *end) {
+			G_LOG(ERR,
+				"gatekeeper: log file mode \"%s\" is not a number\n",
+				arg);
+			return ARGP_ERR_UNKNOWN;
```

Comment:
At this point, the log has not been set up; in fact, we are gathering the parameters to set up the log. Use ARGP facility, that it, replace `G_LOG()` and the `return` with `argp_error(state, "the log file mode \"%s\" is not an number", arg);`. Notice that `argp_error()` does not return.

This comment applies to other parts of `parse_opt()`.